include/dmlc/parameter.h

```diff
@@ -853,4 +853,78 @@ inline ValueType GetEnv(const char *key,
   return ret;
 }
 }  // namespace dmlc
+
+// define string serializer for vector, to get the arguments
+namespace std {
```

Comment:
I tried adding this into a new
```
template<>
class FieldEntry<std::vector<int> >
    : public FieldEntryBase<FieldEntry<std::vector<int> >, std::vector<int> > {
```
but for some reason, it kept ignoring this and ran https://github.com/dmlc/dmlc-core/pull/197/commits/1163f51ee47feb370529184e363e407574fe93a1#diff-4b3c194d1c3981ab9fe7006fe3428563R518 instead. I have absolutely no idea why that generic template was taking higher priority than this one.

PS: This is taken from the param.h in XGBoost